### ISSUE TITLE
Random predictions during inference for the same image

### ISSUE THREAD

**Issue by ash1995** [matterport/Mask_RCNN#1209]
I recently trained the Mask RCNN on some satellite images, but during inference mode, I'm getting random predictions for the same set of weights for the same image.

That is to say, for a given set of weights and the same image I'm getting different bounding box and mask predictions.

Edit: This is similar to issue #1146. 

Does anyone have an idea of why this is the case?

The following are the different predictions that the model produced for the same image.

![atlanta_nadir8_catid_10300100023bc100_732701_3730539](https://user-images.githubusercontent.com/32295036/50781411-c2098200-12be-11e9-96d5-c69bb033bcee.jpg)

![atlanta_nadir8_catid_10300100023bc100_732701_3730539_3](https://user-images.githubusercontent.com/32295036/50781432-ccc41700-12be-11e9-89be-3ee9268311f4.jpg)

![atlanta_nadir8_catid_10300100023bc100_732701_3730539_4](https://user-images.githubusercontent.com/32295036/50781454-d6e61580-12be-11e9-84cc-6b9bbe0031b5.jpg)

On some cases, the model produces reasonable predictions like that of the 2nd image but other times it produces completely random results.

I would highly appreciate any insight into this problem.

**Comment by ash1995** [comment]
I think I've solved it. The problem was that I had not loaded the weights of some layers during **inference** time and had excluded them as seen below:

`model.load_weights(model_path, by_name=True, exclude=["mrcnn_bbox_fc", "mrcnn_class_logits", "mrcnn_mask"])`

However it turns out that they were being used and were being initialised randomly. The random initialisation of those weights were the cause of random predictions during inference.

TLDR: 
Don't exclude layers during inference. The code should be as follows:
`model.load_weights(model_path, by_name=True)`

**Comment by grantchenhuarong** [comment]
We also need to train the model to count roofs either.Thanks for your experiences.
